teeworlds7/user_actions.go

```diff
@@ -13,9 +14,15 @@ import (
 // low level access for experts
 // ----------------------------
 
-func (client *Client) SendPacket(packet *protocol7.Packet) error {
-	if packet.Header.Flags.Resend == false && len(packet.Messages) == 0 && len(client.QueuedMessages) == 0 {
-		return fmt.Errorf("Failed to send packet: payload is empty.")
+func (client *Client) SendPacket(packet *protocol7.Packet) (err error) {
+	defer func() {
+		if err != nil {
+			err = fmt.Errorf("failed to send packet: %w", err)
+		}
+	}()
```

Comment:
what is this defer func doing here and why?

Reply:
it's prepending the error message with a common prefix, here `failed to send packet:`